GitUI/Editor/FileViewerInternal.cs

```diff
@@ -173,7 +178,21 @@ public void SetHighlighting(string syntax)
 
         public void SetHighlightingForFile(string filename)
         {
-            IHighlightingStrategy highlightingStrategy = HighlightingManager.Manager.FindHighlighterForFile(filename);
+            IHighlightingStrategy highlightingStrategy;
+
+            if (filename.EndsWith("git-rebase-todo"))
+            {
+                highlightingStrategy = new RebaseTodoHighlightingStrategy(_moduleProvider());
+            }
+            else if (filename.EndsWith("COMMIT_EDITMSG"))
```

Comment:
We used to call `HighlightingManager.Manager.FindHighlighterForFile(filename);` which found a highligher based on the file's extension. However for these files, it's the file name itself that determines the syntax.

```
# for interactive rebase scripts
C:/.../.git/rebase-merge/git-rebase-todo

# for commit and squash commit message editing
C:/.../.git/COMMIT_EDITMSG
```